### PR TITLE
Added coveo integration and search provider switching

### PR DESCRIPTION
**Pull request by long-wan-ep**
Description:

- Moved Algolia search and searchbar to their own components.
- Added Coveo components:
  - search results page with facets, sorting
  - search bar with suggestions
- Added Coveo credentials to config.
- Search and searchbar components will now render Algolia or Coveo based on the `searchProvider` config.

1080p
image

ipad
image

iphoneX
image

Please reach out to me if you'd like the Coveo credentials to our test account.

Linting:

- [x] No linting errors

Tests:

- [x] Manual tests
- [x] Accessibility tests (no new react-axe errors in console)

Documentation:

- [x] Requires documentation updates

Accounts needed:
- [Coveo account](https://www.coveo.com/en/products/platform)

Configuration parameter changes:
|  Parameter| Importance|Type|Description|
|--|--|--|--|
|`algoliaAppId`| Required for Algolia| String| Algolia application identifier.|
|`algoliaApiKey`| Required for Algolia| String| Algolia API key used to read records.|
|`algoliaPlacesAppId`| Required for Algolia| String| Algolia Places application identifier used for address lookup.|
|`algoliaPlacesApiKey`| Required for Algolia| String| Algolia Places API key used for address lookup.|
|`algoliaIndexName`| Required for Algolia| String| Name of Algolia index used for search functions.|
|`coveoOrg`| Required for Coveo| String| The Coveo organization identifier.|
|`coveoApiKey`| Required for Coveo| String| Coveo API key used to read records.|
|`coveoSourceName`| Required for Coveo| String| The Coveo source name to read records from.|
|`searchProvider`| Required| String| The search provider to use, can be `algolia` or `coveo`.|

Coveo Documentation Links:
- [Installation, tutorials, core concepts, how-tos, etc.](https://docs.coveo.com/en/375/javascript-search-framework/use-the-coveo-javascript-search-framework)
- [Reference for components, classes, interfaces, top-level functions, etc.](https://coveo.github.io/search-ui/globals.html)

### CODE REVIEW

File: src/Search.tsx [hunk 1]
```diff
@@ -1,62 +1,29 @@
...
-  );
+  const renderSearch = () => {
+    switch(config.searchProvider) {
+      case 'coveo':
+        return (<CoveoSearch />)
+      case 'algolia':
+        return (<AlgoliaSearch />)
+      default:
+        return (
+          <p>Unsupported search provider</p>
```
Comment (shaunmaharaj) [inline src/Search.tsx:19]: Please move this text to the messages.json for languages

Comment (long-wan-ep) [reply src/Search.tsx]: Moved that message to locales jsons.

**Review by shaunmaharaj — APPROVED** [review]

**Comment by shaunmaharaj** [comment]
@long-wan-ep Great work on this and thank you for contributing!! We can merge this one once conflicts are resolved 👍

**Comment by long-wan-ep** [comment]
Conflicts resolved.

**Pull request merged**